plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -371,12 +440,304 @@ public synchronized Optional<ConnectorOutputMetadata> finishCreateTable(Connecto
+            return type.getLong(block, position);
+        }
+        if (javaType == double.class) {
+            return type.getDouble(block, position);
+        }
+        if (javaType == Slice.class) {
+            return type.getSlice(block, position);
+        }
+        return type.getObject(block, position);
+    }
+
+    private static String quoted(String name)
+    {
+        return "\"" + name.replace("\"", "\"\"") + "\"";
+    }
+
+    private List<String> getDictionaryPredicates(SchemaTableName tableName, TableInfo info, Map<String, Long> distinctValues, Map<String, Object> minimums, Map<String, Object> maximums)
+    {
+        long schemaDictionarySize = (long) getSchema(tableName.getSchemaName()).properties().getOrDefault(SchemaInfo.MAX_DICTIONARY_SIZE, maxDictionarySize);
+        long maxDictionarySize = (long) info.properties().getOrDefault(TableInfo.MAX_DICTIONARY_SIZE, schemaDictionarySize);
+        if (maxDictionarySize == 0) {
+            return List.of();
+        }
+        List<ColumnInfo> columns = info.columns();
+        Map<String, Type> types = columns.stream().collect(toMap(ColumnInfo::name, ColumnInfo::type));
+        List<String> dictionaryColumns = distinctValues.entrySet().stream()
+                .filter(entry -> entry.getValue() <= maxDictionarySize)
+                .map(Map.Entry::getKey)
+                .collect(toImmutableList());
+        ImmutableMap<String, FakerColumnHandle> columnHandles = info.columns().stream()
+                .collect(toImmutableMap(ColumnInfo::name, ColumnInfo::handle));
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        try (FakerPageSource pageSource = new FakerPageSource(
+                faker,
+                random,
+                dictionaryColumns.stream()
+                        .map(columnHandles::get)
+                        .collect(toImmutableList()),
+                TupleDomain.withColumnDomains(dictionaryColumns.stream()
+                        .collect(toMap(
+                                columnHandles::get,
+                                column -> Domain.create(
+                                        ValueSet.ofRanges(Range.range(
+                                                types.get(column),
+                                                minimums.get(column),
+                                                true,
+                                                maximums.get(column),
+                                                true)),
+                                        false)))),
+                0,
+                maxDictionarySize)) {
+            // TODO support reading values from multiple pages
+            Page page = null;
+            while (page == null) {
+                page = pageSource.getNextPage();
+            }
+            checkState(maxDictionarySize <= page.getPositionCount(), "Max dictionary size cannot be greater than %d".formatted(page.getPositionCount()));
```

Comment:
Should I add a `@Max()` in the configuration and schema/table property validators?

Reply:
If you can express upper bound as a constant - sure.